lua/barbar/render.lua

```diff
@@ -702,59 +550,73 @@ local function generate_tabline(bufnrs, refocus)
   -- Create actual tabline string
   local result = ''
 
-  -- Add offset filler & text (for filetree/sidebar plugins)
+  -- Left offset
   if state.offset.left.width > 0 then
-    --- @type barbar.render.group
-    local offset = {hl = hl_tabline(state.offset.left.hl or 'BufferOffset'), text = ' ' .. state.offset.left.text}
-    local offset_available_width = state.offset.left.width - 2
+    local hl = wrap_hl(state.offset.left.hl)
+    local offset_groups = {
+      { hl = hl, text = (' '):rep(state.offset.left.width) }
+    }
 
-    result = result ..
-      groups_to_string(slice_groups_right({offset}, offset_available_width)) ..
-      (' '):rep(offset_available_width - strwidth(state.offset.left.text) + 1)
-  end
+    local content = {
+      { hl = hl, text = state.offset.left.text }
+    }
+    local content_max_width = state.offset.left.width - 2
+
+    offset_groups =
+      groups.insert(
+        offset_groups,
+        1,
+        groups.slice_right(content, content_max_width))
 
-  --- The highlight of the buffer tabpage fill
-  local hl_buffer_tabpage_fill = hl_tabline('BufferTabpageFill')
+    result = result .. groups.to_string(offset_groups)
+  end
```

Comment:
This seems longer but we're making use of `groups.insert`, which is a magical function that merges groups inside another set of groups at a certain position, taking care of any slicing necessary. We should use it instead of manually slicing. The slicing functions are mainly there (in the render code) to crop content.